packages/flame_3d/lib/src/resources/shader/shader.dart

```diff
@@ -10,6 +10,8 @@ import 'package:flutter_gpu/gpu.dart' as gpu;
 ///
 /// {@endtemplate}
 class ShaderResource extends Resource<gpu.Shader> {
```

Comment:
with the new Resource design there is no longer a need to separate ShaderResource and Shader for testing purposes

one would be able to create a "Shader" in tests no problem as long as they don't call the `.resource` getter

if this new pattern is accepted, I can followup with updating this to be simpler as it was before